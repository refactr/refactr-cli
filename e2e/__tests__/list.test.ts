import { executeWithStdoutOnly } from '../helpers/execute';
import { loadFixtures } from './fixtures';
import { withAddress, withFormat } from '../helpers/options';

beforeAll(async () => {
  return await loadFixtures();
});

describe('refactrctl list', () => {
  test('throws on missing subcommand', async () => {
    await expect(executeWithStdoutOnly(['list'])).rejects.toMatchSnapshot();
  });

  describe('credentials', () => {
    test('throws on missing arguments', async () => {
      await expect(
        executeWithStdoutOnly(['list', 'credentials'])
      ).rejects.toMatchSnapshot();
    });

    test('should work (default formatting)', async () => {
      await expect(
        executeWithStdoutOnly(
          withAddress([
            'list',
            'credentials',
            '--project-id',
            '5ffc715d04ea5ca71d201bb3'
          ])
        )
      ).resolves.toMatchSnapshot();
    });

    test('should work (--format=table)', async () => {
      await expect(
        executeWithStdoutOnly(
          withAddress([
            'list',
            'credentials',
            '--project-id',
            '5ffc715d04ea5ca71d201bb3'
          ])
        )
      ).resolves.toMatchSnapshot();
    });

    test('should work (--format=json)', async () => {
      await expect(
        executeWithStdoutOnly(
          withFormat(
            withAddress([
              'list',
              'credentials',
              '--project-id',
              '5ffc715d04ea5ca71d201bb3'
            ]),
            'json'
          )
        )
      ).resolves.toMatchSnapshot();
    });

    test('should work (--format=yaml)', async () => {
      await expect(
        executeWithStdoutOnly(
          withFormat(
            withAddress([
              'list',
              'credentials',
              '--project-id',
              '5ffc715d04ea5ca71d201bb3'
            ]),
            'yaml'
          )
        )
      ).resolves.toMatchSnapshot();
    });
  });

  describe('jobs', () => {
    test('throws on missing arguments', async () => {
      await expect(
        executeWithStdoutOnly(['list', 'jobs'])
      ).rejects.toMatchSnapshot();
    });
  });

  describe('pipeline-revisions', () => {
    test('throws on missing arguments', async () => {
      await expect(
        executeWithStdoutOnly(['list', 'pipeline-revisions'])
      ).rejects.toMatchSnapshot();
    });
  });

  describe('pipelines', () => {
    test('throws on missing arguments', async () => {
      await expect(
        executeWithStdoutOnly(['list', 'pipelines'])
      ).rejects.toMatchSnapshot();
    });
  });

  describe('runners', () => {
    test('throws on missing arguments', async () => {
      await expect(
        executeWithStdoutOnly(['list', 'runners'])
      ).rejects.toMatchSnapshot();
    });
  });

  describe('runs', () => {
    test('throws on missing arguments', async () => {
      await expect(
        executeWithStdoutOnly(['list', 'runs'])
      ).rejects.toMatchSnapshot();
    });
  });
});
