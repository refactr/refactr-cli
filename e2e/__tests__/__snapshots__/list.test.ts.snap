// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`refactrctl list credentials should work (--format=json) 1`] = `
"undefined
[
    {
        \\"_id\\": \\"5ffc71a504ea5c3fab201bb8\\",
        \\"project_id\\": \\"5ffc715d04ea5ca71d201bb3\\",
        \\"created\\": \\"2021-01-11T15:41:25.326Z\\",
        \\"name\\": \\"Test CLI Credential 2\\",
        \\"id\\": \\"cli_test_cred_2\\",
        \\"type\\": \\"generic\\"
    },
    {
        \\"_id\\": \\"5ffc718704ea5c7701201bb6\\",
        \\"project_id\\": \\"5ffc715d04ea5ca71d201bb3\\",
        \\"created\\": \\"2021-01-11T15:40:55.242Z\\",
        \\"name\\": \\"Test CLI Credential 1\\",
        \\"id\\": \\"cli_test_cred_1\\",
        \\"type\\": \\"username_password\\"
    }
]
"
`;

exports[`refactrctl list credentials should work (--format=table) 1`] = `
"undefined
[31m _id                      [39m[31m created             [39m[31m name                  [39m[31m id              [39m[31m type              [39m[31m data [39m
 5ffc71a504ea5c3fab201bb8  01/11/2021, 8:41 PM  Test CLI Credential 2  cli_test_cred_2  generic            -    
 5ffc718704ea5c7701201bb6  01/11/2021, 8:40 PM  Test CLI Credential 1  cli_test_cred_1  username_password  -    
"
`;

exports[`refactrctl list credentials should work (--format=yaml) 1`] = `
"undefined
-
    _id: 5ffc71a504ea5c3fab201bb8
    project_id: 5ffc715d04ea5ca71d201bb3
    created: '2021-01-11T15:41:25.326Z'
    name: Test CLI Credential 2
    id: cli_test_cred_2
    type: generic
-
    _id: 5ffc718704ea5c7701201bb6
    project_id: 5ffc715d04ea5ca71d201bb3
    created: '2021-01-11T15:40:55.242Z'
    name: Test CLI Credential 1
    id: cli_test_cred_1
    type: username_password

"
`;

exports[`refactrctl list credentials should work (default formatting) 1`] = `
"undefined
[31m _id                      [39m[31m created             [39m[31m name                  [39m[31m id              [39m[31m type              [39m[31m data [39m
 5ffc71a504ea5c3fab201bb8  01/11/2021, 8:41 PM  Test CLI Credential 2  cli_test_cred_2  generic            -    
 5ffc718704ea5c7701201bb6  01/11/2021, 8:40 PM  Test CLI Credential 1  cli_test_cred_1  username_password  -    
"
`;

exports[`refactrctl list credentials throws on missing arguments 1`] = `
"refactrctl credentials [options]

Options:
      --version     Show version number  [boolean]
  -v, --verbose     Print detailed output  [boolean]
      --format      Specifies output format  [choices: \\"wide\\", \\"json\\", \\"yaml\\"] [default: \\"wide\\"]
      --address     Address of the Refactr API server. This can also be specified via the REFACTR_ADDRESS environment variable. Defaults to https://api.refactr.it/v1  [string]
      --auth-token  Authentication token. This can also be specified via the REFACTR_AUTH_TOKEN environment variable  [string]
      --help        Show help  [boolean]
      --filter      Filter output using JsonPath  [string]
      --project-id  ID of project containing the credentials  [string] [required]
      --type  [string] [choices: \\"api_token\\", \\"aws_access_key\\", \\"azure_service_principal\\", \\"bearer_token\\", \\"generic\\", \\"ssh_key\\", \\"username_password\\", \\"vault_app_role\\"]
      --sort        Sort order  [string] [choices: \\"created_asc\\", \\"created_desc\\"]
      --limit  [number]
      --offset  [number]

Missing required argument: project-id
"
`;

exports[`refactrctl list jobs throws on missing arguments 1`] = `
"refactrctl jobs [options]

Options:
      --version     Show version number  [boolean]
  -v, --verbose     Print detailed output  [boolean]
      --format      Specifies output format  [choices: \\"wide\\", \\"json\\", \\"yaml\\"] [default: \\"wide\\"]
      --address     Address of the Refactr API server. This can also be specified via the REFACTR_ADDRESS environment variable. Defaults to https://api.refactr.it/v1  [string]
      --auth-token  Authentication token. This can also be specified via the REFACTR_AUTH_TOKEN environment variable  [string]
      --help        Show help  [boolean]
      --filter      Filter output using JsonPath  [string]
      --project-id  ID of the project containing the jobs  [string] [required]

Missing required argument: project-id
"
`;

exports[`refactrctl list pipeline-revisions throws on missing arguments 1`] = `
"refactrctl runs [options]

Options:
      --version      Show version number  [boolean]
  -v, --verbose      Print detailed output  [boolean]
      --format       Specifies output format  [choices: \\"wide\\", \\"json\\", \\"yaml\\"] [default: \\"wide\\"]
      --address      Address of the Refactr API server. This can also be specified via the REFACTR_ADDRESS environment variable. Defaults to https://api.refactr.it/v1  [string]
      --auth-token   Authentication token. This can also be specified via the REFACTR_AUTH_TOKEN environment variable  [string]
      --help         Show help  [boolean]
      --filter       Filter output using JsonPath  [string]
      --project-id   ID of the project containing the pipeline revision  [string] [required]
      --pipeline-id  ID of the pipeline containing the pipeline revision  [string] [required]

Missing required arguments: project-id, pipeline-id
"
`;

exports[`refactrctl list pipelines throws on missing arguments 1`] = `
"refactrctl list pipelines

List pipelines for a project

Options:
      --version     Show version number  [boolean]
  -v, --verbose     Print detailed output  [boolean]
      --format      Specifies output format  [choices: \\"wide\\", \\"json\\", \\"yaml\\"] [default: \\"wide\\"]
      --address     Address of the Refactr API server. This can also be specified via the REFACTR_ADDRESS environment variable. Defaults to https://api.refactr.it/v1  [string]
      --auth-token  Authentication token. This can also be specified via the REFACTR_AUTH_TOKEN environment variable  [string]
      --help        Show help  [boolean]
      --filter      Filter output using JsonPath  [string]
      --project-id  ID of the project containing the pipelines  [string] [required]
      --sort        Order in which returned pipelines should be sorted  [choices: \\"modified_asc\\", \\"modified_desc\\", \\"name_asc\\", \\"name_desc\\"]
      --limit  [number]
      --offset  [number]
      --search  [string]

Missing required argument: project-id
"
`;

exports[`refactrctl list runners throws on missing arguments 1`] = `
"refactrctl list runners

List runners for an organization

Options:
      --version          Show version number  [boolean]
  -v, --verbose          Print detailed output  [boolean]
      --format           Specifies output format  [choices: \\"wide\\", \\"json\\", \\"yaml\\"] [default: \\"wide\\"]
      --address          Address of the Refactr API server. This can also be specified via the REFACTR_ADDRESS environment variable. Defaults to https://api.refactr.it/v1  [string]
      --auth-token       Authentication token. This can also be specified via the REFACTR_AUTH_TOKEN environment variable  [string]
      --help             Show help  [boolean]
      --filter           Filter output using JsonPath  [string]
      --project-id       ID of the project containing the runners  [string]
      --organization-id  ID of the organization containing the runners  [string]

Missing required argument: either organization-id or project-id
"
`;

exports[`refactrctl list runs throws on missing arguments 1`] = `
"refactrctl runs [options]

Options:
      --version      Show version number  [boolean]
  -v, --verbose      Print detailed output  [boolean]
      --format       Specifies output format  [choices: \\"wide\\", \\"json\\", \\"yaml\\"] [default: \\"wide\\"]
      --address      Address of the Refactr API server. This can also be specified via the REFACTR_ADDRESS environment variable. Defaults to https://api.refactr.it/v1  [string]
      --auth-token   Authentication token. This can also be specified via the REFACTR_AUTH_TOKEN environment variable  [string]
      --help         Show help  [boolean]
      --filter       Filter output using JsonPath  [string]
      --project-id   ID of the organization containing the runs  [string] [required]
      --pipeline-id  ID of a pipeline used to filter the runs  [string]
      --job-id       ID of a job used to filter the runs  [string]

Missing required argument: project-id
"
`;

exports[`refactrctl list throws on missing subcommand 1`] = `
"Usage: refactrctl list <command> [options]

Commands:
  refactrctl list credentials         List credentials from a project
  refactrctl list projects            List projects from an organization
  refactrctl list organizations       List organizations one belongs to
  refactrctl list jobs                List jobs for a project
  refactrctl list runs                List runs for a project
  refactrctl list pipeline-revisions  List runs for a project
  refactrctl list runners             List runners for an organization
  refactrctl list pipelines           List pipelines for a project

Options:
      --version     Show version number  [boolean]
  -v, --verbose     Print detailed output  [boolean]
      --format      Specifies output format  [choices: \\"wide\\", \\"json\\", \\"yaml\\"] [default: \\"wide\\"]
      --address     Address of the Refactr API server. This can also be specified via the REFACTR_ADDRESS environment variable. Defaults to https://api.refactr.it/v1  [string]
      --auth-token  Authentication token. This can also be specified via the REFACTR_AUTH_TOKEN environment variable  [string]
      --help        Show help  [boolean]
      --filter      Filter output using JsonPath  [string]

Command must be specified
"
`;
